File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    memo = {}
    
    for i in completion:
        if i in memo:
            memo[i]+=1
        else:
            memo[i]=1
            
    for i in participant:
        if i in memo:
            if memo[i] == 0:
                return i
            else:
                memo[i]-=1
        else:
            return i

    return 0
    

    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    participant.sort();
    completion.sort();
    answer=participant.pop();
    for(let i=0;i<completion.length;i++){
        if(participant[i]!==completion[i]){
            answer=participant[i];
            break;
        } 
    }
    
    return answer;
}